docs/data/data-grid/column-definition/column-definition.md

```diff
@@ -0,0 +1,148 @@
+
+The following are the native column types:
+
+- `'string'` (default)
+- `'number'`
+- `'date'`
+- `'dateTime'`
+- `'boolean'`
+- `'singleSelect'`
+- `'actions'`
+
+### Converting types
+
+Default methods, such as filtering and sorting, assume that the type of the values will match the type of the column specified in `type` (e.g. the values of a `number` column will be numbers).
+For example, values of column with `type: 'dateTime'` are expecting to be stored as a `Date()` objects.
+If for any reason, your data type is not the correct one, you can use `valueGetter` to parse the value to the correct type.
+
+```tsx
+{
+  field: 'lastLogin',
+  type: 'dateTime',
+  valueGetter: ({ value }) => value && new Date(value),
+}
+```
+
+### Special properties
+
+To use most of the column types, you only need to define the `type` property in your column definition.
+However, some types require additional properties to be set to make them work correctly:
+
+- If the column type is `'singleSelect'`, you also need to set the `valueOptions` property in the respective column definition. These values are options used for filtering and editing.
+
+  ```tsx
+  {
+    field: 'country',
+    type: 'singleSelect',
+    valueOptions: ['United Kingdom', 'Spain', 'Brazil']
+  }
+  ```
+
+  > ⚠️ When using objects values for `valueOptions` you need to provide `value` and `label` fields for each option: `{ value: string, label: string }`
+
+- If the column type is `'actions'`, you need to provide a `getActions` function that returns an array of actions available for each row (React elements).
+  You can add the `showInMenu` prop on the returned React elements to signal the data grid to group these actions inside a row menu.
+
+  ```tsx
+  {
+    field: 'actions',
+    type: 'actions',
+    getActions: (params: GridRowParams) => [
+      <GridActionsCellItem icon={...} onClick={...} label="Delete" />,
+      <GridActionsCellItem icon={...} onClick={...} label="Print" showInMenu />,
+    ]
+  }
+  ```
+
+{{"demo": "ColumnTypesGrid.js", "bg": "inline"}}
+
+## Custom column types
+
+You can extend the native column types with your own by simply spreading the necessary properties.
+
+The demo below defines a new column type: `usdPrice` that extends the native `number` column type.
+
+```ts
+const usdPrice: GridColTypeDef = {
+  type: 'number',
+  width: 130,
+  valueFormatter: ({ value }) => valueFormatter.format(Number(value)),
+  cellClassName: 'font-tabular-nums',
+};
+```
+
+{{"demo": "CustomColumnTypesGrid.js", "bg": "inline"}}
+
+> ⚠ If an unsupported column type is used the `string` column type will be used instead.
+
+## Column menu
+
+By default, each column header displays a column menu. The column menu allows actions to be performed in the context of the target column, e.g. filtering. To disable the column menu, set the prop `disableColumnMenu={true}`.
+
+{{"demo": "ColumnMenuGrid.js", "bg": "inline"}}
```

Comment:
This section could move to the "Column header" page

Reply:
Done